.circleci/config.yml

```diff
@@ -1,34 +1,55 @@
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: circleci/ruby:2.5.1-node-browsers
+      environment:
+        RAILS_ENV: test
+        RACK_ENV: test
+
 version: 2
 jobs:
   build:
-    parallelism: 1
-    working_directory: ~/allocation-manager
-    docker:
-      - image: circleci/ruby:2.5.1-node-browsers
-        environment:
-          RAILS_ENV: test
-          RACK_ENV: test
+    <<: *defaults
     steps:
       - checkout
+      - attach_workspace:
+          at: ~/repo
       - run:
           name: Which bundler?
           command: bundle -v
+      - restore_cache:
+          keys:
+            - allocation-manager-allocation-manager-v1-{{ checksum "Gemfile.lock" }}
+            - allocation-manager-allocation-manager-v1-
       - run:
           name: Update npm
           command: 'sudo npm install -g npm@latest'
       - run:
           name: Install GOV.UK frontend modules
           command: npm install
-      - restore_cache:
-          keys:
-            - allocation-manager-v1-{{ checksum "Gemfile.lock" }}
-            - allocation-manager-v1-
-      - run: gem install did_you_mean -v 1.2.1
       - run: bundle check --path vendor/bundle || bundle install --path vendor/bundle
       - save_cache:
           key: allocation-manager-v1-{{ checksum "Gemfile.lock" }}
           paths:
+            - ~/repo/vendor/bundle
+      - persist_to_workspace:
+          root: .
+          paths: 
             - vendor/bundle
+            - node_modules/
+
+  test:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/repo
+      - run: bundle --path vendor/bundle
+      - run:
+          name: Download Code Climate
+          command: |
+            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
```

Comment:
Can the code climate tool be persisted to the workspace, or is it already cached?